tsdb/engine/tsm1/compact_test.go

```diff
@@ -3896,6 +3896,204 @@ func TestDefaultPlanner_Plan_ForceFull(t *testing.T) {
+		},
+		{
+			blockCounts: []int{
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultAggressiveMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+			},
+			optimizedName: "02-05.tsm",
+		},
+		{
+			blockCounts: []int{
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultAggressiveMaxPointsPerBlock,
+			},
+			optimizedName: "04-04.tsm",
+		},
+	}
+
+	ffs := &fakeFileStore{
+		PathsFn: func() []tsm1.FileStat {
+			return testSet
+		},
+	}
+	cp := tsm1.NewDefaultPlanner(ffs, tsdb.DefaultCompactFullWriteColdDuration)
+
+	e := MustOpenEngine(tsdb.InmemIndexName)
+	e.CompactionPlan = cp
+	e.Compactor = tsm1.NewCompactor()
+	e.Compactor.FileStore = ffs
+
+	fileGroup := make([]string, 0, len(testSet))
+	for j := 0; j < len(testSet); j++ {
+		fileGroup = append(fileGroup, testSet[j].Path)
+	}
+
+	for i := 0; i < len(blockCounts); i++ {
+		require.NoError(t, ffs.SetBlockCounts(blockCounts[i].blockCounts), "failed setting block counts")
+		ok, fName, _ := e.IsGroupOptimized(fileGroup)
+		require.Equal(t, blockCounts[i].optimizedName != "", ok, "unexpected result for optimization check")
+		require.Equal(t, blockCounts[i].optimizedName, fName, "unexpected file name in optimization check")
+	}
+}
+
+func TestEnginePlanCompactions(t *testing.T) {
+	testFileSets := [][]tsm1.FileStat{
+		[]tsm1.FileStat{
+			{
+				Path: "01-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "02-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "03-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "04-04.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+		},
+		[]tsm1.FileStat{
+			{
+				Path: "01-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "02-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "03-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+			{
+				Path: "04-05.tsm",
+				Size: 256 * 1024 * 1024,
+			},
+		},
+	}
+	testBlockCountsAndResults := []struct {
+		blockCounts      []int
+		level1Groups     []tsm1.CompactionGroup
+		level2Groups     []tsm1.CompactionGroup
+		level3Groups     []tsm1.CompactionGroup
+		level4Groups     []tsm1.CompactionGroup
+		level5Groups     []tsm1.CompactionGroup
+		level5Aggressive bool
+	}{
+		{
+			blockCounts: []int{
+				tsdb.DefaultAggressiveMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+			},
+			level5Groups: []tsm1.CompactionGroup{
+				{"01-05.tsm", "02-05.tsm", "03-05.tsm", "04-04.tsm"},
+			},
+			level5Aggressive: false,
+		},
+		{
+			blockCounts: []int{
+				tsdb.DefaultAggressiveMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+				tsdb.DefaultMaxPointsPerBlock,
+			},
+			level5Groups: []tsm1.CompactionGroup{
+				{"01-05.tsm", "02-05.tsm", "03-05.tsm", "04-05.tsm"},
+			},
+			level5Aggressive: false,
```

Comment:
I believe that this should be true, but we get 0 generations back from `PlanOptimize` because we skip files in use.  Please step through the second run of this test and see if you agree.

Reply:
I believe this should be `true` since we have a file that is at the aggressive block count within these TSM groups. I think there are two things that would need to adjusted. 
1. https://github.com/influxdata/influxdb/pull/26319/files#diff-e0d6c334f09b26e18b81206ad3b242f2fd1ef09a7b23ea326a13eea3ac916a9aR2306-R2309 We will need to `Release` the acquired TSM group here that is already in the `filesInUse` map so they can be re-acquired by the PlanOptimize planner run. 
2. https://github.com/influxdata/influxdb/pull/26319/files#diff-e0d6c334f09b26e18b81206ad3b242f2fd1ef09a7b23ea326a13eea3ac916a9aR2322 This test will need to account for a group that has a TSM file that is already at `aggressiveCompactionPointsPerBlock`